rpc-csharp/src/Runtime/MessageDispatcher.cs

```diff
@@ -7,15 +7,34 @@
 
 namespace rpc_csharp
 {
-    public class AckHelper
+    public struct ParsedMessage
     {
+        public RpcMessageTypes messageType;
+        public object message;
+        public uint messageNumber;
+
+        public ParsedMessage(RpcMessageTypes messageType, object message, uint messageNumber)
+        {
+            this.messageType = messageType;
+            this.message = message;
+            this.messageNumber = messageNumber;
+        }
+    }
+
+    public class MessageDispatcher
+    {
+        public static int ID = 0;
+        public int _id = 0;
+        public event Action<ParsedMessage> OnParsedMessage;
+
         private readonly Dictionary<string, (Action<StreamMessage>, Action<Exception>)> oneTimeCallbacks =
             new Dictionary<string, (Action<StreamMessage>, Action<Exception>)>();
 
-        private readonly ITransport transport;
+        public readonly ITransport transport;
```

Comment:
do we need this as public?

Reply:
> do we need this as public?

Yes. Sometimes we need to access the transport directly using the MessageDispatcher.